worker/upload.go

```
@@ -569,6 +569,8 @@ func (mgr *uploadManager) Upload(ctx context.Context, r io.Reader, contracts []a
 		select {
 		case <-mgr.shutdownCtx.Done():
 			return false, "", errWorkerShutDown
+		case <-ctx.Done():
+			return false, "", ctx.Err()
```

I'd return a custom error here. Like `errUploadInterrupted`.

---

👍  good idea